superset/viz.py

```diff
@@ -553,6 +554,10 @@ def json_data(self) -> str:
         return json.dumps(self.data)
 
 
+QUERY_MODE_RAW = "raw"
+QUERY_MODE_AGGREGATE = "aggregate"
```

Comment:
These should probably be placed in `utils/core.py` as
```python
class QueryMode(str, Enum):
    RAW = "raw"
   AGGREGATE = "aggregate"
```
I would also suggest adding a deprecated type for detecting query mode, which would look at the presence of `all_columns`, and flag as `raw` is present, and `aggregate` if not.

Reply:
Yeah, I was unsure about how to do string enum in Python, so went with the easiest route.

Should we have a `core/consts.py` file?

Reply:
> I would also suggest adding a deprecated type for detecting query mode (`AUTO`?), which would look at the presence of `all_columns`, and flag as `raw` is present, and `aggregate` if not.

The logic is there, it's just not its own query mode. Kind of feel it'd be more useful to add a `QueryMode.from_str(str)` method than introducing an `auto` mode.

Reply:
> Yeah, I was unsure about how to do string enum in Python, so went with the easiest route.
> 
> Should we have a `core/consts.py` file?

Oh yes, it's been put off for too long. I think it's more pythonic to call it `constants.py`. @john-bodley any strong feelings of how we could consolidate constants, enums and perhaps type aliases in one easy to find place? Having constants under `utils/core.py` is slightly confusing.

Anyway, I suggest not doing that refactor in this PR, let's do that separately. So I propose just throwing this under `utils/core.py` for now.

Reply:
There is a `superset/contants.py` file, but nothing much seems going on there.

Just added a `QueryMode` in `utils/core.py` as well as a `Enum.get` method to safeguard invalid enum values.